python/lsst/meas/astrom/match.py

```diff
@@ -0,0 +1,102 @@
+     - @ref meas_astrom_match_IO
+     - @ref meas_astrom_match_Config
+     - @ref meas_astrom_match_Example
+
+    @section meas_astrom_match_Purpose  Description
+
+    Match sources to reference objects. This is often done as a preliminary step to fitting an astrometric
+    or photometric solution. For details about the matching algorithm see match.h
+
+    @section meas_astrom_match_Initialize   Task initialisation
+
+    @copydoc \_\_init\_\_
+
+    @section meas_astrom_match_IO       Invoking the Task
+
+    @copydoc run
+
+    @section meas_astrom_match_Config       Configuration parameters
+
+    See @ref MatchConfig
+
+    @section meas_astrom_match_Example  A complete example of using MatchTask
+
+    config = MatchConfig()
+    task = MatchTask(butler=butler, config=config)
+    matchResults = task.run(catalog)
+
+    """
+
+    ConfigClass = MatchConfig
+
+    def __init__(self, butler=None, refObjLoader=None, **kwargs):
+        """!Ctor
+
+        Either a 'butler' or 'refObjLoader' is required.
+
+        @param butler  Data butler, or None
+        @param refObjLoader  For loading reference objects (lsst.meas.algorithms.LoadReferenceObjectsTask), or
+            None
+        @param kwargs  Other keyword arguments required for instantiating a Task (e.g., 'config')
+        """
+        Task.__init__(self, **kwargs)
+        if not refObjLoader:
+            self.makeSubtask("refObjLoader", butler=butler)
+        else:
+            self.refObjLoader = refObjLoader
+
+    def run(self, catalog, filterName=None):
+        """!Load reference objects and match to them
+
+        @param catalog[in]  Catalog to match to (lsst.afw.table.SourceCatalog)
+        @param filterName[in]  Name of filter, for loading fluxes (str)
+        @return Struct with matches (lsst.afw.table.SourceMatchVector) and
+            matchMeta (lsst.meas.astrom.MatchMetadata)
+        """
+        circle = self.calculateCircle(catalog)
+        matchMeta = MatchMetadata(circle.center, circle.radius, filterName)
+        refData = self.refObjLoader.loadSkyCircle(circle.center, circle.radius, filterName)
+        matches = afwTable.matchRaDec(refData.refCat, catalog, self.config.matchRadius*arcseconds)
+        self.log.info("Matched %d from %d input and %d reference sources" %
+                      (len(matches), len(catalog), len(refData.refCat)))
+        return Struct(matches=matches, matchMeta=matchMeta)
+
+    def calculateCircle(self, catalog):
+        """!Calculate a circle enclosing the catalog
+
+        @param catalog[in]  Catalog we will encircle (lsst.afw.table.SourceCatalog)
+        @return Struct with center (lsst.afw.coord.Coord) and radius (lsst.afw.geom.Angle)
+        """
+        coordList = [src.getCoord() for src in catalog]
+        center = afwCoord.averageCoord(coordList)
+        radius = max(center.angularSeparation(coord) for coord in coordList)
+        return Struct(center=center, radius=radius)
```

Comment:
I think this is the best we can do given the available API for the `refObjLoader`.  It may not be the most efficient algorithm but I don't think it's far off if the sources in the `catalog` are constrained to be in the same area, like they are if they're all from the same image; and that's my use case.  If not, the user may have to write and `retarget` a more efficient implementation.